src/utilities/routerNavigation.browser.spec.ts

```diff
@@ -16,59 +15,35 @@ test('when go is called, forwards call to window history', () => {
 })
 
 test('when back is called, forwards call to window history', () => {
-  vi.spyOn(window.history, 'back')
+  vi.spyOn(window.history, 'go')
 
   const onAfterLocationUpdate = vi.fn()
   const history = createRouterNavigation({ onAfterLocationUpdate })
 
   history.back()
 
-  expect(window.history.back).toHaveBeenCalledOnce()
+  expect(window.history.go).toHaveBeenCalledOnce()
 })
 
 test('when forward is called, forwards call to window history', () => {
-  vi.spyOn(window.history, 'forward')
+  vi.spyOn(window.history, 'go')
 
   const onAfterLocationUpdate = vi.fn()
   const history = createRouterNavigation({ onAfterLocationUpdate })
 
   history.forward()
 
-  expect(window.history.forward).toHaveBeenCalledOnce()
-})
-
-test('when update is called, calls updateBrowserUrl', () => {
-  vi.spyOn(utilities, 'updateBrowserUrl')
-
-  const onAfterLocationUpdate = vi.fn()
-  const url = random.number().toString()
-  const history = createRouterNavigation({ onAfterLocationUpdate })
-
-  history.update(url)
-
-  expect(utilities.updateBrowserUrl).toHaveBeenCalledWith(url, undefined)
+  expect(window.history.go).toHaveBeenCalledOnce()
 })
 
-test('when update is called and same origin calls onAfterLocationUpdate', async () => {
-  vi.spyOn(utilities, 'isSameOrigin').mockReturnValue(true)
-
+test('when update is called executes hooks', async () => {
+  const onBeforeLocationUpdate = vi.fn(() => Promise.resolve(true))
   const onAfterLocationUpdate = vi.fn()
   const url = random.number().toString()
-  const history = createRouterNavigation({ onAfterLocationUpdate })
+  const history = createRouterNavigation({ onBeforeLocationUpdate, onAfterLocationUpdate })
 
   await history.update(url)
 
-  expect(onAfterLocationUpdate).toHaveBeenCalledWith(url)
-})
-
-test('when update is called and not same origin does not call onAfterLocationUpdate ', () => {
-  const onAfterLocationUpdate = vi.fn()
-  vi.spyOn(utilities, 'isSameOrigin').mockReturnValue(true)
-
-  const url = random.number().toString()
-  const history = createRouterNavigation({ onAfterLocationUpdate })
-
-  history.update(url)
-
-  expect(onAfterLocationUpdate).not.toHaveBeenCalled()
```

Comment:
I feel like this test was just broken anyway. Because it was mocking the return value of `isSameOrigin` to `true` but the test says its testing when NOT same origin. 

Anyway from what I can tell the history library will automatically call `window.location.assign` if its not same origin. Will probably want to verify that

Reply:
should we write a test for that? Probably more of an "ingetegration test"

Reply:
possibly though I'm not 100% sure how to write that accurately in a emulated browser environment.